receiver/awsfirehosereceiver/internal/unmarshaler/otlpmetricstream/unmarshaler_test.go

```diff
@@ -4,117 +4,84 @@
 package otlpmetricstream
 
 import (
-	"slices"
+	"context"
+	"encoding/binary"
 	"testing"
-	"time"
 
-	"github.com/gogo/protobuf/proto"
 	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/pdata/pcommon"
-	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
-	"go.uber.org/zap"
+	"go.opentelemetry.io/collector/pdata/pmetric"
 
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsfirehosereceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
 )
 
-func TestType(t *testing.T) {
-	unmarshaler := NewUnmarshaler(zap.NewNop(), component.NewDefaultBuildInfo())
-	require.Equal(t, TypeStr, unmarshaler.Type())
+func TestNewUnmarshaler(t *testing.T) {
+	unmarshaler, err := NewUnmarshaler(context.Background())
+	require.NoError(t, err)
+	require.NotNil(t, unmarshaler)
 }
 
-func createMetricRecord() []byte {
-	er := pmetricotlp.NewExportRequest()
-	rsm := er.Metrics().ResourceMetrics().AppendEmpty()
-	sm := rsm.ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
-	sm.SetName("TestMetric")
-	dp := sm.SetEmptySummary().DataPoints().AppendEmpty()
-	dp.SetCount(1)
-	dp.SetSum(1)
-	qv := dp.QuantileValues()
-	minQ := qv.AppendEmpty()
-	minQ.SetQuantile(0)
-	minQ.SetValue(0)
-	maxQ := qv.AppendEmpty()
-	maxQ.SetQuantile(1)
-	maxQ.SetValue(1)
-	dp.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+// getMetricRecord reads the resource metrics inside file
+// and adds a uvarint prefix to it, before returning the
+// byte array.
+func getMetricRecord(t *testing.T, file string) []byte {
+	metrics, err := golden.ReadMetrics(file)
+	require.NoError(t, err)
 
-	temp, _ := er.MarshalProto()
-	record := proto.EncodeVarint(uint64(len(temp)))
-	record = append(record, temp...)
+	m := pmetric.ProtoMarshaler{}
+	data, err := m.MarshalMetrics(metrics)
+	require.NoError(t, err)
+
+	buf := make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(buf, uint64(len(data)))
+	record := buf[:n]
+	record = append(record, data...)
 	return record
 }
 
 func TestUnmarshal(t *testing.T) {
-	unmarshaler := NewUnmarshaler(zap.NewNop(), component.NewDefaultBuildInfo())
+	t.Parallel()
+
+	unmarshaler, errU := NewUnmarshaler(context.Background())
+	require.NoError(t, errU)
+
 	testCases := map[string]struct {
-		record             []byte
-		wantResourceCount  int
-		wantMetricCount    int
-		wantDatapointCount int
-		wantErr            string
+		record                  []byte
+		expectedMetricsFilename string
+		wantErr                 string
```

Comment:
These are the same unit tests as before. I have just put them in files so we can compare the metrics to each other, instead of the length of metrics / data points.

Reply:
We have coverage in the encoding extension, do we even need all these tests? I would be inclined to delete the internal unmarshaler and use the encoding extension directly, and add a test at the top level to make sure the encoding is wired up correctly. Just enough to very it can decode OTLP metrics -- it doesn't need to cover all the paths of the encoding extension.

Reply:
I agree that it is redundant, I kept it to make the change easier. I will remove it.